src/Catel.MVVM/Catel.MVVM.Shared/Collections/Extensions/SuspensionContextExtensions.cs

```diff
@@ -36,51 +35,79 @@ public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateEventArg
                 return ArrayShim.Empty<NotifyRangedCollectionChangedEventArgs>();
             }
 
-            // Determine creation depending on mode
-            ICollection<NotifyRangedCollectionChangedEventArgs> eventArgsList = new List<NotifyRangedCollectionChangedEventArgs>();
-            switch (mode)
+            if (SuspensionContext<T>.EventsGeneratorsRegistry.Count == 0)
```

Comment:
As this is called in all code paths put this to the initalization of the property: At the ctor or with a additional extension method. Because this method is not about initalization the SuspensionContext class.